de_templates/ingest/Ingest_SourceType_SourceName/data_factory/adf_datasets.tf.jinja

```diff
@@ -0,0 +1,11 @@
+resource "azurerm_data_factory_custom_dataset" "ds_{{ dataset_name }}" {
+  name            = "ds_{{ dataset_name }}"
+  data_factory_id = data.azurerm_data_factory.factory.id
+  type            = "{% if data_source_type == 'azure_sql' %}AzureSqlTable{% else %}''{% endif %}"
```

Comment:
There are a few places where we use `if data_source_type == 'azure_sql'` in the templates to convert values. Thinking if these grow to handle more data source types these statements will not be nice to handle. 
Is it possible to pull out conversion of values for a `data_source_type` into a separate method that can populate additional parameters, then the templates don't need the `if` statements?

Reply:
If that's not easily do-able, an alternative could be defining a new param at the top of the template file (like where we've used `{% set pipeline_name = 'Ingest_' ~ dataset_name -%}`), just so all the conditional logic is defined at the start of the template

Reply:
I am aware of this. I looked into it briefly before and didn't find an obvious solution to set jinja parameters in one place globally except for doing it in the data CLI and passing these parameters together with the config. I thought that this would make the templates less transparent and I avoided it. Another option is to set these parameters once per a template (similar to the `pipeline_name`), but it still would have to be done at the top of each template that uses them.

Reply:
I see you added a comment while I was responding. I guess we have two options:
- set it at the top of the templates similar to `pipeline_name`
- move this logic to Data CLI and pass together with the config

Maybe there is a better approach, but I couldn't easily find it, and my experience with Jinja is not too big yet :)

Reply:
done